### PR TITLE
feat: add plugin surface size property and sizeChanged signal

### PR DESCRIPTION
**Pull request by yixinshark**
as title

Log: as title
Influence: tray display

### CODE REVIEW

File: panels/dock/pluginmanagerextension_p.h [hunk 1]
```diff
@@ -105,6 +108,7 @@ class PluginSurface : public QWaylandShellSurfaceTemplate<PluginSurface>, public
     uint32_t m_flags;
     uint32_t m_pluginType;
     uint32_t m_sizePolicy;
+    QSize m_size;
```
Comment (18202781743) [inline panels/dock/pluginmanagerextension_p.h:111]: 这个不需要吧, 直接调用bufferSize ?

File: panels/dock/pluginmanagerextension.cpp [hunk 2]
```diff
@@ -28,6 +28,15 @@ PluginSurface::PluginSurface(PluginManager* manager, const QString& pluginId, co
     init(resource.resource());
     setExtensionContainer(surface);
     QWaylandCompositorExtension::initialize();
+
+    connect(surface, &QWaylandSurface::bufferSizeChanged, this, [this, surface](){
```
Comment (18202781743) [inline panels/dock/pluginmanagerextension.cpp:32]: sizeChanged可以直接连接bufferSizeChanged

**Review by 18202781743 — APPROVED** [review]

File: panels/dock/pluginmanagerextension_p.h (hunk 1 above)
Comment (yixinshark) [reply panels/dock/pluginmanagerextension_p.h]: 1

File: panels/dock/pluginmanagerextension.cpp (hunk 2 above)
Comment (yixinshark) [reply panels/dock/pluginmanagerextension.cpp]: 1

**Pull request merged**